src/atomate2/vasp/flows/defect.py

```diff
@@ -0,0 +1,331 @@
+"""Flows used in the calculation of defect properties."""
+
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass, field
+from typing import Iterable
+
+from jobflow import Flow, Job, Maker, OutputReference, job
+from numpy.typing import NDArray
+from pymatgen.analysis.defect.core import Defect, get_sc_fromstruct
+from pymatgen.analysis.defect.generators import DefectGenerator
+from pymatgen.core.structure import Structure
+
+from atomate2.vasp.jobs.base import BaseVaspMaker
+from atomate2.vasp.jobs.core import RelaxMaker, StaticMaker
+from atomate2.vasp.jobs.defect import (
+    FiniteDifferenceMaker,
+    collect_outputs,
+    get_ccd_documents,
+    perform_defect_calculations,
+    spawn_energy_curve_calcs,
+)
+from atomate2.vasp.schemas.defect import CCDDocument
+from atomate2.vasp.sets.core import StaticSetGenerator
+from atomate2.vasp.sets.defect import AtomicRelaxSetGenerator
+
+logger = logging.getLogger(__name__)
+
+################################################################################
+# Default settings                                                            ##
+################################################################################
+
+DEFAULT_DISTORTIONS = (-1, -0.15, -0.1, -0.05, 0, 0.05, 0.1, 0.15, 1)
+DEFECT_INCAR_SETTINGS = {
+    "ISMEAR": 0,
+    "SIGMA": 0.05,
+    "KSPACING": None,
+    "ENCUT": 500,
+}
+DEFECT_KPOINT_SETTINGS = {"reciprocal_density": 64}
+
+DEFECT_RELAX_GENERATOR: AtomicRelaxSetGenerator = AtomicRelaxSetGenerator(
+    use_structure_charge=True,
+    user_incar_settings=DEFECT_INCAR_SETTINGS,
+    user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
+)
+DEFECT_STATIC_GENERATOR: StaticSetGenerator = StaticSetGenerator(
+    user_incar_settings=DEFECT_INCAR_SETTINGS,
+    user_kpoints_settings=DEFECT_KPOINT_SETTINGS,
+)
+
+
+################################################################################
+# Formation Energy                                                            ##
+################################################################################
+
+
+@dataclass
+class FormationEnergyMaker(Maker):
+    """Class to generate VASP input sets for the calculation of the formation energy diagram.
+
+    Parameters
+    ----------
+    name: str
+        The name of the flow created by this maker.
+    relax_maker: .BaseVaspMaker or None
+        A maker to perform a atomic-position-only relaxation on the defect charge states.
+        If None, the defaults will be used.
+    """
+
+    name: str = "formation energy"
+    relax_maker: BaseVaspMaker = field(
+        default_factory=lambda: RelaxMaker(
+            input_set_generator=DEFECT_RELAX_GENERATOR,
+        )
+    )
+
+    def make(self, defect_gen: DefectGenerator, sc_mat: NDArray | None = None):
```

Comment:
`defect_gen` and `sc_mat` could be output references if you were using this maker in the middle of flow (likely to happen since you'll want to do a structural relaxation etc first).

For that reason, it might make sense to put almost all of this logic inside `perform_defect_calculations`.

Reply:
I think everything in the flow is wrapped in a job now.